packages/pigeon/lib/dart_generator.dart

```diff
@@ -92,6 +92,24 @@ void _writeCodec(Indent indent, String codecName, Api api) {
   });
 }
 
+/// Creates a Dart type where all type arguments are [Objects].
+String _makeGenericTypeArguments(TypedEntity entity, String nullTag) {
+  return entity.typeArguments != null
+      ? '${entity.baseName}<${entity.typeArguments!.map((TypeDeclaration e) => 'Object$nullTag').reduce(_commaJoin)}>'
```

Comment:
I've provided plausible justification why `reduce` is preferred to `join`, it is logically correct, not in violation of any style guide and not in violation of any established norm in the codebase.  I think this comes down to author's discretion.  I appreciate your thoroughness and it  was an interesting academic question but I suspect you share my feeling that the time this question has occupied has surpassed its usefulness.  Feel free to reach out on VC if that isn't the case.

Reply:
> it was an interesting academic question

I was not trying to have an academic discussion, but to review this PR.

We can discuss offline how to move forward.

Reply:
Sure, reach out when you get around.  Since we spent time discussing it I captured my beef with `Iterable<T>.join` into an issue for Dart: https://github.com/dart-lang/sdk/issues/46871   I'm not sure it will lead to any change but might help us avoid the situation in the future and might be a good example for implementing type constraints for extension methods.

Reply:
Ok, since you are serious about your concerns here I chatted with @cbracken whose been on the Dart team a long time.  He agreed that `reduce` is more type-safe than `join` but he said `join` is more idiomatic and if given the choice he would chose that.  So, I'll conceded the point, `join` it is.  Keep an eye out, I've done this downstream too I'll try to replace them there too.  Done.